lib/data/wrapper.js

```diff
@@ -137,6 +132,16 @@ const data = {
     // replace this
     batchDelete: (locations, log) => {
         async.eachLimit(locations, 5, (loc, next) => {
```

Comment:
I would do the log at line 134 before the async.eachLimit and log the full array so that we know what is intended to be deleted before any possible crash on the multiple async calls.  So no need to log in the async.eachLimit function.